packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -211,10 +221,30 @@ export class TokenDetectionController extends BaseController<
     }, this.config.interval);
   }
 
+  private getCorrectChainId(networkClientId?: NetworkClientId) {
+    if (networkClientId) {
+      return this.getNetworkClientById(networkClientId).configuration.chainId;
+    }
+    return this.config.chainId;
+  }
+
+  _executePoll(
+    networkClientId: string,
+    options: { address: string },
+  ): Promise<void> {
+    return this.detectTokens(networkClientId, options?.address);
+  }
+
   /**
    * Triggers asset ERC20 token auto detection for each contract address in contract metadata on mainnet.
+   *
+   * @param networkClientId - The ID of the network client to use.
+   * @param accountAddress - The account address to use.
    */
-  async detectTokens() {
+  async detectTokens(
+    networkClientId?: NetworkClientId,
+    accountAddress?: string,
```

Comment:
Using positional arguments for multiple optional arguments seems like it would make it possible to use this method awkwardly (e.g. `detectTokens(undefined, '0x12345')`. What do you feel about putting these in an (optional) options object?